### PR TITLE
docs: getting started notes on self-signed cert (#9429)

### LINKED ISSUE #9429: CLI returns "certificate signed by unknown authority"
Checklist:

* [ ] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [ ] I've included steps to reproduce the bug.
* [ ] I've pasted the output of `argocd version`.

**Describe the bug**

I am unable to establish a secure connection to ArgoCD:
```shell
$ argocd login argocd-server
WARNING: server certificate had error: x509: certificate signed by unknown authority. Proceed insecurely (y/n)? n
```

**To Reproduce**
On a clean installation of Docker Desktop:

1. Install nginx-ingress
```shell
helm repo add ingress-nginx https://kubernetes.github.io/ingress-nginx
```
2. Install argocd:
```shell
kubectl apply -n argocd -f https://raw.githubusercontent.com/argoproj/argo-cd/stable/manifests/install.yaml
```
3.  Create the ingress:
Save this:
```yaml
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: argocd-server-ingress
  namespace: argocd
  annotations:
    kubernetes.io/ingress.class: nginx
    nginx.ingress.kubernetes.io/force-ssl-redirect: "true"
    nginx.ingress.kubernetes.io/ssl-passthrough: "true"
spec:
  rules:
  - host: argocd-server
    http:
      paths:
      - path: /
        pathType: Prefix
        backend:
          service:
            name: argocd-server
            port: 
               number: 443
```
as argocd-ingress.yml and then apply:
```shell
kubectl apply -n argocd -f argocd-ingress.yml
```
4. Ensure that argocd-server is in localhost:
```shell
grep argocd-server /etc/hosts || echo '127.0.0.1     argocd-server' | sudo tee -a /etc/hosts
```
5. Test:
```shell
argocd login argocd-server
WARNING: server certificate had error: x509: certificate signed by unknown authority. Proceed insecurely (y/n)? n
``` 

Checking with openssl indicates that the cert is coming from ArgoCD:
```shell
$ echo quit | openssl s_client -showcerts -servername argocd-server -connect argocd-server:443 | head
depth=0 O = Argo CD
verify error:num=18:self signed certificate
verify return:1
depth=0 O = Argo CD
verify return:1
CONNECTED(00000003)
---
Certificate chain
 0 s:O = Argo CD
   i:O = Argo CD
-----BEGIN CERTIFICATE-----
MIIDYjCCAkqgAwIBAgIQXxa6rX0ZcIgeHI4opRxuyzANBgkqhkiG9w0BAQsFADAS
MRAwDgYDVQQKEwdBcmdvIENEMB4XDTIyMDUxNzEyMDc1M1oXDTIzMDUxNzEyMDc1
M1owEjEQMA4GA1UEChMHQXJnbyBDRDCCASIwDQYJKoZIhvcNAQEBBQADggEPADCC
AQoCggEBAOhItSIG+Xa60BZQ+NupQ3TV7W1HPJbi3uH4Xk8hs3KBlIlGOlLgiPBt
DONE
```

I have also tried using the argocd-server-tls secret to validate that the certificate being used is not the nginx-ingress one, and that produces errors that include the valid names of the cert, so the cert is definitely coming from the argocd pod - but the argocd CLI doesn't accept it.

I cannot find any documentation about what the cert is _supposed_ to be.
At the moment the [FAQ entry for this](https://argo-cd.readthedocs.io/en/stable/faq/#why-am-i-getting-x509-certificate-signed-by-unknown-authority-when-using-the-cli) contains no more information than the CLI output.
It would be really helpful if that entry contained details on what the cert should be.

**Expected behavior**
Some sort of request for credentials, anything other than a cert error.

**Version**

```shell
$ argocd version
argocd: v2.3.3+07ac038.dirty
  BuildDate: 2022-03-30T05:17:09Z
  GitCommit: 07ac038a8f97a93b401e824550f0505400a8c84e
  GitTreeState: dirty
  GoVersion: go1.18
  Compiler: gc
  Platform: linux/amd64
FATA[0000] Argo CD server address unspecified
```
Running on Windows 11 using WSL2.

### PR DESCRIPTION
**Pull request by YaytayAtWork**
The default certificate created by Argo CD is a standard self-signed certificate.
There is nothing wrong with this, but it means that the instructions in the Getting Started guide cannot be followed.
This PR adds a note to point users in the correct direction to set up a secure cert (or to use --insecure on all the examples in Getting Started).

Signed-off-by: Jim Talbut <jim.talbut@groupgti.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
One nitpick. Thanks!

File: docs/getting_started.md [hunk 1]
```diff
@@ -29,6 +29,10 @@ kubectl create namespace argocd
 kubectl apply -n argocd -f https://raw.githubusercontent.com/argoproj/argo-cd/stable/manifests/core-install.yaml
 ```
 
+This default installation will have a self-signed certificate and cannot be accessed securely.
```
[the remark this answers is not in the document]
Comment (YaytayAtWork) [reply docs/getting_started.md:32]: Hmm, so really there are three options:
1. Your own cert.
2.  Add the self-signed cert.
3. --insecure.

How about:
This default installation will have a self-signed certificate and cannot be accessed without a bit of extra work.
Do one of:

-  Follow the [instructions to configure a certificate](./operator-manual/tls) (and ensure that the client OS trusts it).
-  Configure the client OS to trust the self signed certificate.
-  Use the --insecure flag on all Argo CD CLI operations in this guide.



Accurate and complete, but a bit long winded.

**Review by jannfis — APPROVED** [review]
LGTM. Thank you!

File: docs/getting_started.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (crenshaw-dev) [reply docs/getting_started.md]: I like it!

**Comment by YaytayAtWork** [comment]
Not sure what I need to do to unblock the merge.

**Comment by crenshaw-dev** [comment]
@YaytayAtWork nothing, I was just being slow. 🙂 lgtm, thanks again!

**Review by crenshaw-dev — APPROVED** [review]

**Comment by Yaytay** [comment]
Does anyone know why the list hasn't rendered correctly in the latest docs (https://argo-cd.readthedocs.io/en/latest/getting_started/)?
Should I have left a space before the first entry?

**Pull request merged**